valkey-k8s-operator

```diff
@@ -0,0 +1,267 @@
+
+```yaml
+apiVersion: cache.percona.com/v1alpha1
+kind: Valkey
+metadata:
+  name: my-valkey
+spec:
+  mode: cluster
+  replicas: 3
+  sentinel:
+    enabled: true
+  metrics:
+    enabled: true
+  tls:
+    enabled: true
+  persistence:
+    enabled: true
+    type: both
+    storageClass: fast-disks
+    size: 5Gi
+  config:
+    configMapName: custom-valkey-conf
+  modules:
+    - name: libvalkey_bloom.so
+      path: /opt/valkey/modules/libvalkey_bloom.so
+```
+
+Operator Behavior
+
+- Deploys Bitnami Valkey chart under the hood.
+- Supports 2 modes:
+  - Standalone: 1 primary, optional replicas.
+  - Sentinel: quorum-based HA with automatic failover.
+- Exposes Valkey connection info as a Kubernetes `Secret`.
+- Handles TLS setup via cert-manager or self-signed certs.
+- Optionally loads custom config via ConfigMap.
+- Optional metrics sidecar for Prometheus scraping.
+
+Configuration
+
+| Feature             | Config Options                                                                 |
+|---------------------|--------------------------------------------------------------------------------|
+| Deployment Mode     | `spec.sentinel.enabled: true/false`                                           |
+| TLS                 | `spec.tls.enabled: true` with cert-manager integration                         |
+| Persistence         | `spec.persistence.type: rdb/aof/both`, `storageClass`, `size`                 |
+| Modules             | `spec.modules`: list of `.so` modules to include                               |
+| Monitoring          | `spec.metrics.enabled: true`                                                   |
+| Custom Config       | `spec.config.configMapName`                                                   |
+| Pod Customization   | node selectors, tolerations, resource limits, securityContext, etc.           |
+
+Secret Output
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: valkey-test-binding
+type: Opaque
+stringData:
+  host: valkey-test.svc
+  port: "6379"
+  password: ********
+  sentinelHost: valkey-test.svc
+  sentinelPort: "26379"
+  tlsEnabled: "true"
+```
+
+Authentication and Authorization
+
+- No RBAC-specific changes are needed outside standard Kubernetes role management.
+- Valkey credentials are stored in Kubernetes Secrets.
+- TLS certs issued via cert-manager will follow existing RBAC rules.
+
+RDB
+
+- Persistence modes include:
+  - `rdb`: periodic snapshotting
+  - `aof`: append-only logging
+  - `both`: hybrid approach (default for production)
+
+Configuration (Runtime)
+
+- Users can override default Helm values using CR fields or a `valkey.conf` file via ConfigMap.
+- Sensitive data like passwords are not stored in the config map and are auto-generated and stored in Secrets.
+
+Keyspace notifications
+
+N/A – this Operator does not modify Valkey internals; only orchestrates deployment.
+
+Cluster mode
+
+- Sentinel mode uses quorum-based failover.
+- Standalone mode supports scaling replicas.
```

Comment:
Sentinel and standalone are not cluster mode.

IIUC, we want three different deployment styles:

1. Standalone (a single node, or with replicas but no automatic failover)
2. Sentinel
3. Cluster

Ah, I see "Custer mode" is a heading from the template. Let's ignore the template. :)